azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -23,30 +23,6 @@ func resourceArmKubernetesCluster() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
-		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
-			if v, exists := diff.GetOk("network_profile"); exists {
-				rawProfiles := v.([]interface{})
-				if len(rawProfiles) == 0 {
-					return nil
-				}
-
-				// then ensure the conditionally-required fields are set
-				profile := rawProfiles[0].(map[string]interface{})
-				networkPlugin := profile["network_plugin"].(string)
-
-				if networkPlugin == "kubenet" {
-					dockerBridgeCidr := profile["docker_bridge_cidr"].(string)
-					dnsServiceIP := profile["dns_service_ip"].(string)
-					serviceCidr := profile["service_cidr"].(string)
-
-					if dockerBridgeCidr == "" || dnsServiceIP == "" || serviceCidr == "" {
```

Comment:
we should add this validation back in but ensure that if one of the fields is set then the other fields are set too

Reply:
They either need to be all empty or all non-empty, this is server side logic. Although we can add  it here to improve the experience, but we also introduced the risk to guess the server logic. Anyway, I made the change as you like.

---
In reply to: [207461949](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207461949) [](ancestors = 207461949)